File: 645. Set Mismatch/findErrorNums.java
class Solution {
    public int[] findErrorNums(int[] nums) {

        int n = nums.length;
        int freq[] = new int[n + 1];

        for (int val : nums) {
            freq[val]++;
        }

        int dupli = 0, missing = 0;
        for (int i = 1; i < n + 1; i++) {
            if (freq[i] == 2)
                dupli = i;
            if (freq[i] == 0)
                missing = i;
        }
        return new int[] { dupli, missing };
    }
}

File: 645. Set Mismatch/findErrorNums.cpp
class Solution
{
public:
    vector<int> findErrorNums(vector<int> &nums)
    {

        vector<int> ans = {-1, -1};

        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++)
        {
            m[nums[i]]++;
        }
        for (auto it : m)
        {
            if (it.second > 1)
            {
                ans[0] = it.first;
                break;
            }
        }
        int n = nums.size();
        for (int i = 1; i <= n; i++)
        {
            if (m.find(i) == m.end())
            {
                ans[1] = i;
                break;
            }
        }

        return ans;
    }
};